pr_agent/git_providers/local_git_provider.py

```diff
@@ -0,0 +1,201 @@
+import logging
+import uuid
+from collections import Counter
+from pathlib import Path
+from typing import List
+
+from git import GitCommandError, Repo
+
+from pr_agent.config_loader import _find_repository_root, settings
+from pr_agent.git_providers.git_provider import EDIT_TYPE, FilePatchInfo, GitProvider
+
+
+class PullRequestMimic:
+    """
+    This class mimics the PullRequest class from the PyGithub library for the LocalGitProvider.
+    """
+
+    def __init__(self, title: str, diff_files: List[FilePatchInfo]):
+        self.title = title
+        self.diff_files = diff_files
+
+
+class LocalGitProvider(GitProvider):
+    """
+    This class implements the GitProvider interface for local git repositories.
+    It mimics the PR functionality of the GitProvider interface,
+    but does not require a hosted git repository.
+    Instead of providing a PR url, the user provides a local branch path to generate a diff-patch.
+    For the MVP it only supports the /review and /describe capabilities.
+    """
+
+    def __init__(self, branch_name, incremental=False):
```

Comment:
branch_name in the argument (provided from the CLI as --pr_url, to keep compatibility with other providers) is the target branch - usually that will be the main branch. Perhaps I should've called it the target_branch.

repo.head.ref.name would be the name of the head branch. We need it, so we are able to checkout there after we're done with the review (from the temporary branch we create and work on). I explain the idea behind temporary branches in the next comment.

I assumed here we will always want to compare between HEAD..target_branch, but I'm open to the user providing the branch_name instead of HEAD, sure.